Build error: no matching conversion for functional-style cast from 'std::string'

Build fails for version 2.22.1 with the following:

```
src/libcmd/common-eval-args.cc:23:103: error: no matching conversion for functional-style cast from 'std::string' (aka 'basic_string<char, char_traits<char>, allocator<char>>') to 'nix::MixEvalArgs::AutoArgExpr'
        .handler = {[&](std::string name, std::string expr) { autoArgs.insert_or_assign(name, AutoArg{AutoArgExpr(expr)}); }}
```

**Steps To Reproduce**

1. Follow instructions to build from source at https://nixos.org/manual/nix/stable/installation/installing-source
2. Make sure C++ compiler is c++20 as stated in instructions (CXXFLAGS='-std=c++20')
3. Run `gmake`
4. See error

**Additional context**

```
  CXX    src/libcmd/misc-store-flags.o
src/libcmd/common-eval-args.cc:23:103: error: no matching conversion for functional-style cast from 'std::string' (aka 'basic_string<char, char_traits<char>, allocator<char>>') to 'nix::MixEvalArgs::AutoArgExpr'
        .handler = {[&](std::string name, std::string expr) { autoArgs.insert_or_assign(name, AutoArg{AutoArgExpr(expr)}); }}
                                                                                                      ^~~~~~~~~~~~~~~~
src/libcmd/common-eval-args.hh:31:12: note: candidate constructor (the implicit copy constructor) not viable: no known conversion from 'std::string' (aka 'basic_string<char, char_traits<char>, allocator<char>>') to 'const nix::MixEvalArgs::AutoArgExpr' for 1st argument
    struct AutoArgExpr { std::string expr; };
           ^
src/libcmd/common-eval-args.hh:31:12: note: candidate constructor (the implicit move constructor) not viable: no known conversion from 'std::string' (aka 'basic_string<char, char_traits<char>, allocator<char>>') to 'nix::MixEvalArgs::AutoArgExpr' for 1st argument
src/libcmd/common-eval-args.hh:31:12: note: candidate constructor (the implicit default constructor) not viable: requires 0 arguments, but 1 was provided
src/libcmd/common-eval-args.cc:31:100: error: no matching conversion for functional-style cast from 'std::string' (aka 'basic_string<char, char_traits<char>, allocator<char>>') to 'nix::MixEvalArgs::AutoArgString'
        .handler = {[&](std::string name, std::string s) { autoArgs.insert_or_assign(name, AutoArg{AutoArgString(s)}); }},
                                                                                                   ^~~~~~~~~~~~~~~
src/libcmd/common-eval-args.hh:32:12: note: candidate constructor (the implicit copy constructor) not viable: no known conversion from 'std::string' (aka 'basic_string<char, char_traits<char>, allocator<char>>') to 'const nix::MixEvalArgs::AutoArgString' for 1st argument
    struct AutoArgString { std::string s; };
           ^
src/libcmd/common-eval-args.hh:32:12: note: candidate constructor (the implicit move constructor) not viable: no known conversion from 'std::string' (aka 'basic_string<char, char_traits<char>, allocator<char>>') to 'nix::MixEvalArgs::AutoArgString' for 1st argument
src/libcmd/common-eval-args.hh:32:12: note: candidate constructor (the implicit default constructor) not viable: requires 0 arguments, but 1 was provided
src/libcmd/common-eval-args.cc:39:103: error: no matching conversion for functional-style cast from 'std::string' (aka 'basic_string<char, char_traits<char>, allocator<char>>') to 'nix::MixEvalArgs::AutoArgFile'
        .handler = {[&](std::string name, std::string path) { autoArgs.insert_or_assign(name, AutoArg{AutoArgFile(path)}); }},
                                                                                                      ^~~~~~~~~~~~~~~~
src/libcmd/common-eval-args.hh:33:12: note: candidate constructor (the implicit copy constructor) not viable: no known conversion from 'std::string' (aka 'basic_string<char, char_traits<char>, allocator<char>>') to 'const nix::MixEvalArgs::AutoArgFile' for 1st argument
    struct AutoArgFile { std::filesystem::path path; };
           ^
src/libcmd/common-eval-args.hh:33:12: note: candidate constructor (the implicit move constructor) not viable: no known conversion from 'std::string' (aka 'basic_string<char, char_traits<char>, allocator<char>>') to 'nix::MixEvalArgs::AutoArgFile' for 1st argument
src/libcmd/common-eval-args.hh:33:12: note: candidate constructor (the implicit default constructor) not viable: requires 0 arguments, but 1 was provided
  CXX    src/libcmd/repl-interacter.o
  CXX    src/libcmd/repl.o
  CXX    tests/functional/test-libstoreconsumer/main.o
3 errors generated.
gmake: *** [mk/patterns.mk:17: src/libcmd/common-eval-args.o] Error 1
gmake: *** Waiting for unfinished jobs....
rm src/nix/generated-doc/files/profiles.md
```

Add any other context about the problem here.

Compiler is CLANG

--- a/src/libcmd/common-eval-args.cc
+++ b/src/libcmd/common-eval-args.cc
@@ -20,7 +20,7 @@ MixEvalArgs::MixEvalArgs()
         .description = "Pass the value *expr* as the argument *name* to Nix functions.",
         .category = category,
         .labels = {"name", "expr"},
-        .handler = {[&](std::string name, std::string expr) { autoArgs.insert_or_assign(name, AutoArg{AutoArgExpr(expr)}); }}
+        .handler = {[&](std::string name, std::string expr) { autoArgs.insert_or_assign(name, AutoArg{AutoArgExpr{expr}}); }}
     });
 
     addFlag({
@@ -28,7 +28,7 @@ MixEvalArgs::MixEvalArgs()
         .description = "Pass the string *string* as the argument *name* to Nix functions.",
         .category = category,
         .labels = {"name", "string"},
-        .handler = {[&](std::string name, std::string s) { autoArgs.insert_or_assign(name, AutoArg{AutoArgString(s)}); }},
+        .handler = {[&](std::string name, std::string s) { autoArgs.insert_or_assign(name, AutoArg{AutoArgString{s}}); }},
     });
 
     addFlag({
@@ -36,7 +36,7 @@ MixEvalArgs::MixEvalArgs()
         .description = "Pass the contents of file *path* as the argument *name* to Nix functions.",
         .category = category,
         .labels = {"name", "path"},
-        .handler = {[&](std::string name, std::string path) { autoArgs.insert_or_assign(name, AutoArg{AutoArgFile(path)}); }},
+        .handler = {[&](std::string name, std::string path) { autoArgs.insert_or_assign(name, AutoArg{AutoArgFile{path}}); }},
         .completer = completePath
     });
 

